### PR TITLE
feat: create http front cache package

### PR DESCRIPTION
**Pull request by tassioFront**
## Infos

[Task](https://juntossomosmais.monday.com/boards/1757192123/pulses/7708040193?term=cache&termColumns=XQAAAALOAAAAAAAAAABBKoIjccxORhDgvbWQlLB4Ta76NoV2eZts6r3BdP1jnY6JBXHm9V_PdzIR02temtyzHHvFOY60d6i7uUcE00No3QrC1281DVXdqgj-t6kxADPVIN-4fli3K8KLjGC2vqeXvNaoTlPewG2DAK0JHxW5f8FF_-25fi_swK9quIq_nur20Nv__5p1AAA)

#### What is being delivered?

creating HTTP front cache package based on [http-front-cache](https://www.npmjs.com/package/utility-http-front-cache)

#### What impacts?

- new package on the time-out-market repo
- ability to cache HTTP requests on frontend
- `cacheOnSessionStorage` function to cache data on session storage
- `cacheFactory` function to cache data on a given provider (such as session storage, local storage, in-memory etc)
 
#### Reversal plan

reset the main branch on the previous commit and delete the release version

#### Evidences

Using  `cacheOnSessionStorage` to cache [articles](https://frontend-pattern.vercel.app/articles) by a given `userName`

[Screencast from 2024-11-04 10-11-45.webm](https://github.com/user-attachments/assets/2acde4ae-7e29-422c-9ebe-c61bb9c833a0)

### CODE REVIEW

**Comment by tassioFront** [comment]
I created a new key on Sonar for linters to configure the repo as monorepo: juntossomosmais_fe-time-out-market-linters

File: .husky/commit-msg [hunk 1]
```diff
@@ -1,4 +1 @@
-#!/usr/bin/env sh
```
Comment (tassioFront) [inline .husky/commit-msg:1]: it isn't required anymore

File: packages/http-front-cache/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  };
+};
+
+// usage
+const cachedFetchData = cacheOnMyCustomProvider(fetchData, 5 * 60 * 1000); // Cache for 5 minutes
+
+cachedFetchData('exampleParam').then((result) => {
+  console.log(result);
+});
+```
```
Comment (felipefialho) [inline packages/http-front-cache/README.md:96]: 🚀 🚀 🚀

File: packages/http-front-cache/src/lib/cache-factory.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import hash from 'object-hash'
+import pako from 'pako'
+import { CacheEntry, CacheFactory } from './types'
+
+/**
+ * BE CAREFUL ON USE IT!!! Use the following constraints to understand when to use it:
```
Comment (felipefialho) [inline packages/http-front-cache/src/lib/cache-factory.ts:6]: praise: nice warnings

**Review by felipefialho — APPROVED** [review]
👏 👏 👏

**Review by MayaraRMA — APPROVED** [review]

**Pull request merged**